test/performance/odp_packet_gen.c

```diff
@@ -303,8 +304,11 @@ static void print_usage(void)
 	       "                            src_port/dst_port. Comma-separated (no spaces) list of\n"
 	       "                            count values: <src_port count>,<dst_port count>\n"
 	       "                            Default value: 0,0\n"
-	       "  -C, --no_udp_checksum     Do not calculate UDP checksum. Instead, set it to\n"
-	       "                            zero in every packet.\n"
+	       "  -C, --no_udp_checksum     Do not calculate UDP SW checksum. Instead, set it to\n"
+	       "                            zero in every packet, this may be overridden by\n"
+	       "                            '--checksum_offload' option.\n"
+	       "  -X, --checksum_offload    Enable L4 checksum offloads: checksums are checked\n"
+	       "                            at input and inserted at output.\n"
```

Comment:
Maybe there could be separate options for RX and TX checksum offloads. Also, if RX checksum offload is enabled, checksum validity could be checked during packet receive.

Reply:
Changed according to offline discussions in v3. Still just single option, but RX side offloading remains disabled with a warning log if packet I/O does not support also packet dropping on errors.